Sources/TestingMacros/Support/AttributeDiscovery.swift

```diff
@@ -178,23 +188,63 @@ struct AttributeInfo {
       }
     }))
 
-    // Any arguments of the test declaration macro which specify test arguments
-    // need to be wrapped a closure so they may be evaluated lazily by the
-    // testing library at runtime. If any such arguments are present, they will
-    // begin with a labeled argument named `arguments:` and include all
-    // subsequent unlabeled arguments.
-    var otherArguments = self.otherArguments
-    if let argumentsIndex = otherArguments.firstIndex(where: { $0.label?.tokenKind == .identifier("arguments") }) {
-      for index in argumentsIndex ..< otherArguments.endIndex {
-        var argument = otherArguments[index]
-        argument.expression = .init(ClosureExprSyntax { argument.expression.trimmed })
-        otherArguments[index] = argument
+    // If there are any parameterized test function arguments, wrap each in a
+    // closure so they may be evaluated lazily at runtime.
+    if let testFunctionArguments {
+      arguments += testFunctionArguments.map { argument in
+        var copy = argument
+        copy.expression = .init(ClosureExprSyntax { argument.expression.trimmed })
+        return copy
       }
     }
 
-    arguments += otherArguments
     arguments.append(Argument(label: "sourceLocation", expression: sourceLocation))
 
     return LabeledExprListSyntax(arguments)
   }
 }
+
+extension Collection<Argument> {
+  /// This collection of test function arguments augmented with explicit type
+  /// information based on the specified function parameters, if appropriate.
+  ///
+  /// - Parameters:
+  ///   - parameters: The parameters of the function to which this collection of
+  ///     test arguments was passed.
+  ///
+  /// - Returns: An array containing this collection of test arguments with
+  ///   any array literal expressions given explicit type information via an
+  ///   `as ...` cast.
+  fileprivate func testArguments(typedUsingParameters parameters: FunctionParameterListSyntax) -> [Argument] {
+    if count == 1 {
+      let tupleTypes = parameters.lazy
```

Comment:
This will fail to compile if any arguments have specifiers like `borrowing`. (I'm not sure if that's correct in this position.) You should check the exact type of type node.